plugin/pkg/admission/allowunsafedelete/admission.go

```diff
@@ -0,0 +1,141 @@
+	})
+}
+
+// NewAllowUnsafeDelete creates a new admission control handler that will admit
+// an unsafe delete of a corrupt object if the user has the appropriate privilege
+func NewAllowUnsafeDelete() *AllowUnsafeDelete {
+	return &AllowUnsafeDelete{
+		Handler: admission.NewHandler(admission.Delete),
+	}
+}
+
+type AllowUnsafeDelete struct {
+	*admission.Handler
+	authz authorizer.Authorizer
+}
+
+var _ admission.ValidationInterface = &AllowUnsafeDelete{}
+var _ admission.InitializationValidator = &AllowUnsafeDelete{}
+var _ genericadmissioninit.WantsAuthorizer = &AllowUnsafeDelete{}
+
+// SetAuthorizer sets the authorizer.
+func (p *AllowUnsafeDelete) SetAuthorizer(authz authorizer.Authorizer) {
+	p.authz = authz
+}
+
+// ValidateInitialization ensures an authorizer is set.
+func (p *AllowUnsafeDelete) ValidateInitialization() error {
+	if p.authz == nil {
+		return fmt.Errorf("%s requires an authorizer", PluginName)
+	}
+	return nil
+}
+
+// Validate ensures that the user has permission to do 'delete-ignore-read-errors'
+// on the resource being deeted when ignoreStoreReadErrorWithClusterBreakingPotential
+// is enabled, these are the constraints:
+// a) must be a DELETE operation
+// b) ignoreStoreReadErrorWithClusterBreakingPotential is set to true
+// c) the request is a resource, no sub-resource or non-resource endpoint
+// d) the user has permission to do 'delete-ignore-read-errors' on the resource
+func (p *AllowUnsafeDelete) Validate(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	// if the feature is disabled, this plugin, although enabled should not
+	// be active, and requests with the delete option
+	// 'ignoreStoreReadErrorWithClusterBreakingPotential' enabled will
+	// fallback to using the normal deletion flow
+	if !utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AllowUnsafeMalformedObjectDeletion) {
+		return nil
+	}
+	if attr.GetOperation() != admission.Delete || attr.GetOperationOptions() == nil {
+		return nil
+	}
+
+	options, ok := attr.GetOperationOptions().(*metav1.DeleteOptions)
+	if !ok {
+		return fmt.Errorf("expected an option of type: %T, but got: %T", &metav1.DeleteOptions{}, attr.GetOperationOptions())
+	}
+	if !ptr.Deref(options.IgnoreStoreReadErrorWithClusterBreakingPotential, false) {
+		return nil
+	}
+
+	requestInfo, found := request.RequestInfoFrom(ctx)
+	if !found {
+		return admission.NewForbidden(attr, errors.New("no RequestInfo found in the context"))
+	}
+	if !requestInfo.IsResourceRequest || len(attr.GetSubresource()) > 0 {
+		return admission.NewForbidden(attr, errors.New("ignoreStoreReadErrorWithClusterBreakingPotential delete option is not allowed on a subresource or non-resource request"))
+	}
+
+	// if we are here, IgnoreStoreReadErrorWithClusterBreakingPotential
+	// is set to true in the delete options, the user must have permission
+	// to do 'delete-ignore-read-errors' on the given resource.
+	const verb = "delete-ignore-read-errors"
```

Comment:
we probably want the unsafe bit emphasized here because this will be the point where someone might grant access.

Reply:
yes, changed it to `unsafe-delete-ignore-read-errors`